crates/host/src/wasmbus/mod.rs

```diff
@@ -1039,19 +1104,71 @@ impl Lattice {
         }: Command = serde_json::from_slice(payload.as_ref())
             .context("failed to deserialize provider launch command")?;
 
-        debug!(configuration, link_name, provider_ref, "launch provider");
+        debug!(link_name, provider_ref, ?configuration, "launch provider");
 
-        bail!("TODO");
+        let (path, claims) = crate::fetch_provider(&provider_ref, &link_name)
+            .await
+            .context("failed to fetch provider")?;
+        let mut command = process::Command::new(&path);
+
+        // TODO: Publish events
+        match self.providers.write().await.entry(claims.subject.clone()) {
+            hash_map::Entry::Vacant(entry) => {
+                debug!(?path, link_name, "launch a new provider");
+                let child = command
+                    .spawn()
+                    .context("failed to spawn provider process")?;
+                entry.insert(Provider {
+                    claims,
+                    instances: HashMap::from([(
+                        link_name,
+                        ProviderInstance {
+                            child,
+                            id: Ulid::new(),
+                        },
+                    )]),
+                });
+            }
+            hash_map::Entry::Occupied(mut entry) => {
+                debug!(
+                    ?path,
+                    link_name, "launch a new instance of an existing provider"
+                );
+                let Provider { instances, .. } = entry.get_mut();
+                if let hash_map::Entry::Vacant(entry) = instances.entry(link_name) {
+                    let child = command
+                        .spawn()
+                        .context("failed to spawn provider process")?;
+                    entry.insert(ProviderInstance {
+                        child,
+                        id: Ulid::new(),
+                    });
+                }
```

Comment:
There should be an else here, for the error case: the host is already running this provider on this link name

Reply:
> this provider

The host in this branch is not running this provider yet (assuming that the provider claim "subject" field uniquely identifies a provider), so I am not sure what you mean

Reply:
Ah, I see now - I have misinterpreted https://github.com/wasmCloud/wasmcloud-otp/blob/4feeb1ff9b4821690a3e42023a926f4ea5c24ede/host_core/lib/host_core/providers/provider_supervisor.ex#L277-L285

Reply:
I'm talking about a case like this:
```
wash start provider -l foo wasmcloud.azurecr.io/httpserver:0.17.0
wash start provider -l foo wasmcloud.azurecr.io/httpserver:0.17.0
```

The second command should be rejected by the host, since `(httpserver, foo)` is already running